File: 234/solution.py
from random import random
from math import *

upper = 999966663333
p = [False for i in range(1000010)]
cnt = 0
sum = 0


def handle(pi, pj):
    global cnt, sum
    l = pi * pi + 1
    r = min(upper, pj * pj - 1)
    if l > r:
        return
    r += 1
    for i in range(l // pi * pi, r, pi):
        if i < l or i % pj == 0:
            continue
        cnt += 1
        sum += i
    for i in range(l // pj * pj, r, pj):
        if i < l or i % pi == 0:
            continue
        cnt += 1
        sum += i


def Handle():
    last = -1
    for i in range(2, 1000000):
        if not p[i]:
            if last != -1:
                handle(last, i)
            last = i
        for j in range(i << 1, 1000000, i):
            p[j] = True


if __name__ == "__main__":
    Handle()
    print(cnt, sum)

File: 234/solution.cpp
#include<iostream>
#include<math.h>
using namespace std;
const long long upper = 999966663333;
bool p[1000000+10];
long long cnt = 0;
long long sum = 0;
void handle(long long pi, long long pj) {/*{{{*/
	auto l = pi*pi+1;
	auto r = pj*pj-1;
	r = min(r, upper);
	if(l>r) return ;
	// [l, r];
	for(auto i = l/pi*pi;i<=r;i+=pi) {
		if(i<l||i%pj==0) continue;
		cnt++;
		sum+=i;
	}
	for(auto i = l/pj*pj;i<=r;i+=pj) {
		if(i<l||i%pi==0) continue;
		cnt++;
		sum+=i;
	}
}/*}}}*/

void handle() {/*{{{*/
	int last = -1;
	for(int i = 2;i<1000000;i++) {
		if(!p[i]) {
			if(last != -1) handle(last, i);
			last = i;
			for(int j = i<<1;j<1000000;j+=i) p[j] = true;
		}
	}
}/*}}}*/

int main() {
	handle();
	cout<<cnt<<' '<<sum<<endl;
	return 0;
}
